django/contrib/messages/storage/base.py

```diff
@@ -28,11 +56,18 @@ def _prepare(self):
     def __eq__(self, other):
         if not isinstance(other, Message):
             return NotImplemented
-        return self.level == other.level and self.message == other.message
+        return (
+            self.level == other.level
+            and self.message == other.message
+            and self.restrictions == other.restrictions
```

Comment:
If restrictions were made hashable `self.restrictions` could be a `set()`, which would mean that this comparison isn't depending upon the ordering of restrictions in the messages themselves.